src/MutatorOrchestrator.ts

```diff
@@ -6,26 +6,29 @@ import RemoveConditionalsMutator from './mutators/RemoveConditionalsMutator';
 import ReverseConditionalMutator from './mutators/ReverseConditionalMutator';
 import UnaryOperatorMutator from './mutators/UnaryOperatorMutator';
 import {Mutator, StrykerNode, MutatorFactory} from './api/mutant';
+import {Reporter, SourceFile} from './api/report';
 import * as fileUtils from './utils/fileUtils';
 import Mutant from './Mutant';
 import * as parserUtils from './utils/parserUtils';
 import * as log4js from 'log4js';
+import {freezeRecursively} from './utils/objectUtils';
 
 const log = log4js.getLogger('Mutator');
 
 /**
  * Class capable of finding spots to mutate in files.
  */
 export default class MutatorOrchestrator {
-  private mutators: Mutator[];
+  private mutators: Mutator[] = [];
+  private sourceFiles: SourceFile[];
 
-  public constructor() {
+  /**
+   * @param reporter - The reporter to report read input files to
+   */
+  public constructor(private reporter: Reporter) {
```

Comment:
It is an elegant solution but this constructor confused me at first because you told me we were using the `BroadcastReporter` and my mind just went like: *hold on.. is this right?*. But it's correct :) :+1: